types/jasmine-ajax/index.d.ts

```diff
@@ -5,14 +5,10 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-interface JasmineAjaxResponse {
-    status?: number;
-    statusText?: string;
-    responseText?: string;
-    response?: string;
-    contentType?: string;
-    responseHeaders?: { [key: string]: string };
-}
```

Comment:
Is `JasmineAjaxResponse` really *necessarily* the same type as `JasmineAjaxRequestStubReturnOptions`? I mean, I know it *currently* is the same, but is it completely out of the question that Jasmine will some day add an option to `stubRequest.andReturn()` that isn't directly reflected in the fake response?

Reply:
I tried to make the `andReturn` code look like the `andError` code. Both `andError` and `responseError` use a single argument type: `JasmineAjaxRequestStubErrorOptions`.

Actually it's incorrect to use the same type both in the error methods and in the success methods because the `status` field is required both in `respondWith` and `responseError`. But since the original library author violates this requirement in the tests, I've decided to leave it as is.

Reply:
Also there is no response interface in XMLHttpRequest, the response data is stored in the `XMLHttpRequest` object directly. So I believe that having a type called `JasmineAjaxResponse` is a mistake.

But if you don't agree, I'll just make as you ask.

Reply:
I think I'm misunderstanding your arguments, because I don't see how they are related to my question. Has the content of the response object returned by a stub request necessarily the same type as the information passed to `.andReturn`, yes or no?

Reply:
Your question looks incorrect because there is no function that returns a _response_ object in the type definition. I guess you want to know whether the `JasmineAjaxRequest.respondWith` and `JasmineAjaxRequestStub.andReturn` function argument types are the same type or not. If so, then the answer is "no" because the function are independent. But at the moment the types are equal.

Considering this and the way the `JasmineAjaxRequestStubErrorOptions` type is used, I've decided to do this MR the way it's done.

Reply:
Ah, now I see what you mean. Yes, I agree that my question about returning a response object was a bit misguided. I think I assumed that such a thing was going on because of the name `JasmineAjaxResponse`. I now understand it's just the argument type of `respondWith`. Should have checked the code better before asking my question, sorry about that.

Anyway, [the documentation](https://jasmine.github.io/tutorials/mocking_ajax) seems to suggest these types are actually intended to be the same, so you were right to merge them:

> Call `stubRequest` with the url you want to return immediately. Then `andReturn` receives the same type of argument as `respondWith`.